File: sources/Meme 8/ooo/cpacket/ruby/module/mode/Mode.java
package ooo.cpacket.ruby.module.mode;

import net.minecraft.client.Minecraft;
import ooo.cpacket.ruby.module.Module;

public abstract class Mode {
	private Module module;
	private String name;
	
	protected final Minecraft mc = Minecraft.getMinecraft();
	
	public Mode(Module module, String name) {
		this.module = module;
		this.name = name;
	}

	public Module getModule() {
		return this.module;
	}

	public String getName() {
		return this.name;
	}

	public abstract void onEnable();
	
	public abstract void onDisable();
	
}

File: sources/Leaf devEnv/com/leafclient/leaf/utils/client/setting/Mode.kt
package com.leafclient.leaf.utils.client.setting

import com.leafclient.leaf.management.utils.Labelable
import net.minecraft.client.Minecraft

/**
 * A super class used by kill aura modes
 */
abstract class Mode(override val label: String): Labelable {

    protected val mc = Minecraft.getMinecraft()

}